src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs

```diff
@@ -124,7 +124,7 @@ private IChangeToken GetOrAddFilePathChangeToken(string filePath)
                     new[]
                     {
                         changeToken,
-                        new PollingFileChangeToken(new FileInfo(filePath))
+                        new PollingFileChangeToken(new FileInfo(Path.Combine(_root, filePath)))
```

Comment:
This is the code path I found.
PhysicalFilesWatcher.CreateFileChangeToken(string filter) -> GetOrAddChangeToken(filter) -> GetOrAddFilePathChangeToken(filePath)